src/app/components/SubjectHeading/Search/SubjectHeadingSearch.jsx

```diff
@@ -33,14 +33,30 @@ class SubjectHeadingSearch extends React.Component {
 
     this.setState({
       userInput,
-    }, () => {
-      axios(`${appConfig.baseUrl}/api/subjectHeadings/autosuggest?query=${userInput}`)
+    }, this.makeApiCallWithThrottle(this.state.timerId));
+  }
+
+  makeApiCallWithThrottle(timerId) {
+    const apiCall = () => {
+      return axios(`${appConfig.baseUrl}/api/subjectHeadings/autosuggest?query=${this.state.userInput}`)
         .then((res) => {
-          this.setState({
-            suggestions: res.data.autosuggest,
-          });
+          if (res.data.request.query === this.state.userInput) {
+            this.setState({
+              suggestions: res.data.autosuggest,
+              activeSuggestion: 0,
+            });
+          }
         });
-    });
+    };
+
+    if (timerId || this.state.userInput.length === 0) return;
+
+    const newTimerId = setTimeout(() => {
+      apiCall();
+      this.setState({ timerId: undefined });
```

Comment:
Should `apiCall` only execute after `timerId` is set to `undefined`? If so might want to put `apiCall` in the callback to `setState`, i.e. `this.setState({ timerId: undefined }, apiCall)`

Reply:
Good suggestion